jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/caws/DevEnvStatusWatcher.kt

```diff
@@ -0,0 +1,95 @@
+            ?: error("Failed to fetch connection settings from Dev Environment")
+        val envId = System.getenv(CawsConstants.CAWS_ENV_ID_VAR) ?: error("envId env var null")
+        val org = System.getenv(CawsConstants.CAWS_ENV_ORG_NAME_VAR) ?: error("space env var null")
+        val projectName = System.getenv(CawsConstants.CAWS_ENV_PROJECT_NAME_VAR) ?: error("project env var null")
+        val client = connection.awsClient<CodeCatalystClient>()
+        val coroutineScope = projectCoroutineScope(project)
+        coroutineScope.launch(getCoroutineBgContext()) {
+            val initialEnv = client.getDevEnvironment {
+                it.id(envId)
+                it.spaceName(org)
+                it.projectName(projectName)
+            }
+            val inactivityTimeout = initialEnv.inactivityTimeoutMinutes()
+            if (inactivityTimeout == 0) {
+                return@launch
+            }
+            val inactivityTimeoutInSeconds = inactivityTimeout * 60
+            var actualInactivityDuration: Long = 0
+            var lastControllerActivity: Long = 0
+            while (true) {
+                val statusJson = UnattendedStatusUtil.getStatus()
+                val lastActivityTime = statusJson.projects?.first()?.secondsSinceLastControllerActivity ?: 0
+                actualInactivityDuration = if (lastActivityTime > lastControllerActivity) {
+                    lastActivityTime - lastControllerActivity
+                } else {
+                    notifyBackendOfActivity()
+                    lastActivityTime
+                }
+
+                if (actualInactivityDuration >= (inactivityTimeoutInSeconds - 300)) {
+                    try {
+                        val inactivityDurationInMinutes = actualInactivityDuration / 60
+                        val ans = runBlocking {
+                            val continueWorking = withContext(getCoroutineUiContext()) {
+                                return@withContext MessageDialogBuilder.okCancel(
+                                    message("caws.devenv.continue.working.after.timeout.title"),
+                                    message("caws.devenv.continue.working.after.timeout", inactivityDurationInMinutes)
+                                ).ask(project)
+                            }
+                            return@runBlocking continueWorking
+                        }
+
+                        if (ans) {
+                            notifyBackendOfActivity()
+                            lastControllerActivity = actualInactivityDuration
+                        }
+                    } catch (e: Exception) {
+                        notifyError(e.message.toString())
+                    }
+                }
+                Thread.sleep(30000)
+            }
+        }
+    }
+
+    private fun notifyBackendOfActivity() {
+        val request = UpdateActivityRequest(
+            timestamp = System.currentTimeMillis().toString()
+        )
```

Comment:
we don't need this since the environment api will automatically infer the timestamp

Reply:
It does not infer the timestamp correctly, value needs to be provided here